app/models/source.rb

```diff
@@ -6,6 +6,9 @@ class Source < ApplicationRecord
   has_many :application_types, :through => :applications
   has_many :endpoints, :autosave => true, :dependent => :destroy
 
+  attribute :availability_status, :string
+  validates :availability_status, :inclusion => { :in => %w[available unavailable] }, :allow_nil => true
```

Comment:
Since this is going to be an aggregated status (eventually we'd have status on the endpoints and authentications) can we have an intermediate status like mixed or partially_available